config/main.py

```diff
@@ -1824,6 +1824,43 @@ def override_config_by(golden_config_path):
     return
 
 
+def generate_sysinfo(cur_config, config_input, ns=None):
+    # Generate required sysinfo for Golden Config.
+    device_metadata = config_input.get('DEVICE_METADATA')
+
+    if not device_metadata or 'localhost' not in device_metadata:
+        return
+
+    mac = None
+    platform = None
+    cur_device_metadata = cur_config.get('DEVICE_METADATA')
+
+    # Reuse current config's mac and platform. Generate if absent
+    if cur_device_metadata is not None:
+        mac = cur_device_metadata.get('localhost', {}).get('mac')
+        platform = cur_device_metadata.get('localhost', {}).get('platform')
+
+    if not mac or not platform:
```

Comment:
>if not mac or not platform: [](http://example.com/codeflow?start=4&length=27)

Could you separate mac and platform into 2 if-blocks? They seems not relating each other.